### ISSUE TITLE
Updated Replays

### ISSUE THREAD

**Issue by juaoose** [eddy5641/LegendaryClient#93]
I just tested the updated made to the replay system, overall, it improved but you can not yet watch the replays because you get this error when trying to watch the replay:

![error](https://cloud.githubusercontent.com/assets/994594/4887619/25c7d6d0-6386-11e4-8066-1ead719a2b51.png)

Also, the replay tab seems not to be working properly, what I mean is when you provide a username and press download, nothing happens.

The only way to download a replay is from the home page/featured games and even there, the only region that seems to work is NA. (I tried to record a LAN game and it crashes, when I looked at the folder where replays are saved, the name of the replay was XXXXXBR, which may be the problem, since it was supposed to be a LAN game, not BR):

![ee](https://cloud.githubusercontent.com/assets/994594/4887718/20354f44-6387-11e4-949e-c478f1806cff.png)

LOG when trying to record a LAN Featured Game:

```
(03/11/2014 1:25 p. m.) [UNHANDLEDEXCEPTION]: An unhandled exception was thrown
(03/11/2014 1:25 p. m.) [UNHANDLEDEXCEPTION]: No se admiten los formatos de URI.
(03/11/2014 1:25 p. m.) [UNHANDLEDEXCEPTION]:    en System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   en System.IO.Path.GetFullPathInternal(String path)
   en System.IO.Path.GetFullPath(String path)
   en System.Net.WebClient.GetUri(String path)
   en System.Net.WebClient.DownloadFile(String address, String fileName)
   en LegendaryClient.Logic.Replays.ReplayRecorder..ctor(String Server, Int32 GameId, String Region, String Key) en c:\Users\Betty\Desktop\Coding\LegendaryClient\LegendaryClient\Logic\Replays\ReplayRecorder.cs:línea 42
   en LegendaryClient.Windows.MainPage.RecordButton_Click(Object sender, RoutedEventArgs e) en c:\Users\Betty\Desktop\Coding\LegendaryClient\LegendaryClient\Windows\MainPage.xaml.cs:línea 548
   en System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   en System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   en System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   en System.Windows.UIElement.RaiseEvent(RoutedEventArgs e)
   en System.Windows.Controls.Primitives.ButtonBase.OnClick()
   en System.Windows.Controls.Button.OnClick()
   en System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   en System.Windows.UIElement.OnMouseLeftButtonUpThunk(Object sender, MouseButtonEventArgs e)
   en System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   en System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   en System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   en System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   en System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   en System.Windows.UIElement.OnMouseUpThunk(Object sender, MouseButtonEventArgs e)
   en System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   en System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   en System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   en System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   en System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   en System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   en System.Windows.UIElement.RaiseEvent(RoutedEventArgs args, Boolean trusted)
   en System.Windows.Input.InputManager.ProcessStagingArea()
   en System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   en System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   en System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   en System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   en System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   en MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   en MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   en System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   en MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   en System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   en MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   en MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   en System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   en System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   en System.Windows.Threading.Dispatcher.Run()
   en System.Windows.Application.RunDispatcher(Object ignore)
   en System.Windows.Application.RunInternal(Window window)
   en System.Windows.Application.Run(Window window)
   en LegendaryClient.App.Main() en c:\Users\Betty\Desktop\Coding\LegendaryClient\LegendaryClient\obj\Release\App.g.cs:línea 0
```

**MORE NOTES:**
When recording from featured games from each region:
EUNE-> Does not crash-> File name: XXXXXX-EUN1
EUW-> Does not crash-> File name: XXXXXX-EUN1
BR-> Crashes -> File name: XXXXXX-BR1
LAN-> Crashes -> File name: XXXXXX-BR1
LAS-> Crashes -> File name: XXXXXX-BR1
KR-> Crashes -> File name: XXXXXX-KR
NA-> Does not crash-> File name: XXXXXX-NA1
OCE-> Does not crash-> File name: XXXXXX-OC1
Selecting RU made client freeze and then, after selecting record it crashed without creating a replay folder.
# TLDR:
1. Recording only works for some regions and only when done by the FEATURED GAMES menu.
2. The download button on the replay tab does not do anything.
3. Replay server crashes since it does not find all chunks.

Thanks @eddy5641

**Comment by juaoose** [comment]
Managed to make it crash when trying to play a replay, seems like it fails when it does not find all of the chunks (Every recording Ive made seems to lack chunks from 3 to 15-17 )

![repl](https://cloud.githubusercontent.com/assets/994594/4893317/f7f56d28-63bf-11e4-9c8d-655b98c07fe5.png)

'No se pudo encontrar el archivo..' means Could not find or locate the file..

**Comment by eddy5641** [comment]
Usually the replay doesn't work if not recorded before players have entered the game

**Comment by juaoose** [comment]
I tested a game which was like 15s in, Ill try again with a fresh game and update this.

**Comment by eddy5641** [comment]
I do have a replay that I did record. You can get this replay here: https://www.dropbox.com/s/zafwlb6ommodw55/1606934557-NA1.zip?dl=0

**Comment by eddy5641** [comment]
It is a little buggy though.

**Comment by juaoose** [comment]
The one you uploaded kinda works, I think a time limit to record a game would be another nice feature (client wont let you record a game 5 minutes in)

**Comment by eddy5641** [comment]
Will analyze more of lolreplay
